### ISSUE TITLE
Docker-NG does not support selinux in mounted volumes 

### ISSUE THREAD

**Issue by BenjaminSchubert** [saltstack/salt#38368]
### Description of Issue/Question

Docker-NG supports only "ro" and "rw" parameters for mounted volumes.

This can be seen here: https://github.com/saltstack/salt/blob/develop/salt/modules/dockerng.py#L1555

However, docker has officially support for "z" and "Z" mode too (which can be combined with "ro" or "rw". This is to support for selinux relabeling, see https://docs.docker.com/engine/tutorials/dockervolumes/#/volume-labels.

It would be nice for salt to allow for these two, and any combination of them with "ro" and "rw", as without them, it is impossible to run docker on a selinux-enabled host with salt.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

A formula to test this can be : 

```
docker:
  pkg.installed:
    - pkgs:
      - docker
      - python-docker-py

  service.running:
    - name: docker
    - enable: True
    - require:
      - pkg: docker

gitlab docker running:
  dockerng.running:
    - name: gitlab
    - image: gitlab/gitlab-ce

    - require:
      - service: docker

    - port_bindings:
      - "8080:80"
      - "2222:22"
      - "443:443"

    - restart_policy: always

    - binds:
      - {{ pillar["gitlab"]["config"] }}:/etc/gitlab:Z  # notice the :Z
      - {{ pillar["gitlab"]["logs"] }}:/var/log/gitlab:Z  # notice the :Z
      - {{ pillar["gitlab"]["data"] }}:/var/opt/gitlab:Z  # notice the :Z
```

This will fail. Removing `:Z` will work, but then there will be errors on a selinux-enabled host (fedora/centos)

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

- Create a selinux enabled-system (Fedora 25 works well)
- install salt
- add the above formula and apply it
- it will fail because the `:Z` is not recognized
- remove the `:Z`
- the state will work but the image will not be able to read/write from the mounted volumes (see journalctl)

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2016.11.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.3.1611 Core
        machine: x86_64
        release: 3.10.0-514.2.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core
```

**Comment by Ch3LL** [comment]
@BenjaminSchubert looks like you are right as displayed in the code there. We need to add this option to the code. Thanks for the heads up.

**Comment by BenjaminSchubert** [comment]
This has been fixed on the 2017.7 branch and the documentation was merged in https://github.com/saltstack/salt/pull/42016.

I'm closing this as this was decided not to be backported.
